src/main/scala/firrtl/Utils.scala

```diff
@@ -978,6 +978,37 @@ object Utils extends LazyLogging {
     map.view.map({ case (k, vs) => k -> vs.toList }).toList
   }
 
+  /** Combines several separate circuit modules (typically emitted by -e or -p compiler options) into a single circuit */
+  def combine(circuits: Seq[Circuit]): Circuit = {
+    def dedup(modules: Seq[DefModule]): Seq[Either[Module, DefModule]] = {
+      // Left means "lone module", Right means has ExtModules
+      val module: Option[Module] = {
+        val found: Seq[Module] = modules.collect { case m: Module => m }
+        assert(found.size <= 1)
+        found.headOption
+      }
+      val extModules: Seq[ExtModule] = modules.collect { case e: ExtModule => e }.distinct
+
+      if (extModules.isEmpty) Seq(Left(module.get)) else (module ++: extModules).map(Right(_))
+    }
```

Comment:
I think this logic is close but not quite right. Given the following circuit:

```
circuit top :
  module child :
    output out : UInt<8>
    out <= UInt(0)
  module top :
    output out : UInt<8>
    inst c of child
    out <= c.out
```

This if split into multiple files would give:

```
; child.fir
circuit child :
  module child :
    output out : UInt<8>
    out <= UInt(0)
```
```
; top.fir
circuit top :
  extmodule child :
    output out : UInt<8>

  module top :
    output out : UInt<8>
    inst c of child
    out <= c.out
```
Thus when we do the `groupByIntoSeq` we'll have `extmodule child` and `module child` together passed to this function. This function needs to eliminate the `extmodule` by seeing that it's implemented by the associated `Module`.

Now I will grant that down below we have:
```scala
val res = deduped.collect { case Right(m: Module) => m }
```

Which effectively does what I'm saying should be done up here in this function. That is actually wrong because you could truly have an `extmodule` incorrectly getting deleted, consider:

```
circuit top :
  extmodule foo :
    output out : UInt<8>

  module top :
    output out : UInt<8>
    inst c of foo
    out <= c.out
```
I think that the `dedup.collect` below will incorrectly just delete `foo`.